### ISSUE TITLE
Mobile app Synchronization failure when upgrading to newer version

### ISSUE THREAD

**Issue by raghupola-vg** [hzi-braunschweig/SORMAS-Project#5568]
### Bug Description

On the latest build, When a synchronization is performed, it takes too long and throws the Sync fail error. (Only when updated to new build 1.60 to 1.61)
If a fresh install is done, everything works fine.

### Steps to Reproduce

1. Open the sormas mobile app with existing version (1.60) and upgrade to new build (1.61) which is offered.
2. after the upgrade is successful, perform a sync by drag down option

### Expected Behavior

The sync should be successful without any error displayed.

### System Details

* Device: Samsung S10
* SORMAS version: 1.60.0 to 1.61.0
* Android version/Browser: 11
* Server URL: test.sormas.netzlin.com
* User Role: SurvOff

### Additional Information

stacktrace attached
[App sync failure 1.61.0.docx](https://github.com/hzi-braunschweig/SORMAS-Project/files/6560534/App.sync.failure.1.61.0.docx)

**Comment by bernardsilenou** [comment]
@raghupola-vg  @MateStrysewske 
* This issue is one main bug that keeps showing up repeatedly. It would be good for us to have an automatic test to be sure that the mobile app always upgrade or sync with previous versions. We should also drop the condition that mobile users must upgrade to be able to sync with server because it may happen that auto upgrade does not work.

* I currently have a tablet on 1.59.1 and connected to https://release-international.sormas.netzlink.com/sormas-ui/#!about on 1.61.1. When you try to sync, the app asked you to upgrade, but upgrade fails. Thus app cant be used without a fresh installation which then result in data lost.

* The possibility to sync irrespective of the server version and auto update are the two most critical functionalities needed by mobile users.

**Comment by MateStrysewske** [comment]
@bernardsilenou It's technically not possible to allow mobile app users to synchronise with an outdated app version.

**Comment by MateStrysewske** [comment]
@raghupola-vg .docx is a very bad format for error logs, can you please provide them as .txt? Also, are there errors in the server log at the same time? If so, could you provide them as well?

**Comment by MateStrysewske** [comment]
[App Sync failure 1.61.0.txt.zip](https://github.com/hzi-braunschweig/SORMAS-Project/files/6661034/App.Sync.failure.1.61.0.txt.zip)

**Comment by raghupola-vg** [comment]
I do not have access to the server logs.

**Comment by MateStrysewske** [comment]
@raghupola-vg There are two exceptions in your error logs that I can also see in Firebase. One has been resolved by another issue, but the one that probably crashed your synchronization is associated with a very weird state of data on the test instance. Apparently someone pushed over 2,000 cases of the same person (John Doe) to the system; they all have individual clinical courses, but all these clinical courses reference the same health conditions. Your app tried to remove these health conditions from the system (probably because the cases got archived), but since they're referenced in a lot of other cases as well, the database statement failed which leads to an exception.

Bottom line is that this bug is a result of an illegal data state on the test server and therefore doesn't require further action by us at this time.
